core-modules/000QubesVm.py

```diff
@@ -345,7 +346,14 @@ def __init__(self, **kwargs):
                 self.services['qubes-update-check'] = False
 
         # Initialize VM image storage class
-        self.storage = defaults["storage_class"](self)
+        self.storage = get_pool(self.pool_name, self).getStorage()
+        self.dir_path = self.storage.vmdir
+        self.icon_path = os.path.join(self.storage.vmdir, 'icon.png')
+        self.conf_file = os.path.join(self.storage.vmdir, self.name + '.conf')
+        self.appmenus_templates_dir = os.path.join(self.storage.vmdir, 'apps.templates')
+        self.appmenus_icons_dir = os.path.join(self.storage.vmdir, 'apps.icons')
```

Comment:
I was thinking about changing `self.appmenus_templates_dir` (in qubes-core-admin-linux) and others to more like:
```python
def get_appmenus_templates_dir(self):
    return os.path.join(self.storage.vmdir, 'apps.templates')
QubesVM.appmenus_templates_dir = property(get_appmenus_templates_dir)
```
Instead of setting them in `get_attrs_config`.
This way, you don't need `self.storage` at `self.get_attrs_config` time